### DISCUSSION TITLE
Enhancement:  Light on/off

### DISCUSSION THREAD

**Discussion by lanman1** [KlipperScreen/KlipperScreen#215] [Ideas]
Would it be possible to put a light on/off button on the main menu page?  Or even a single button we could assign a macro to do something?  I have a light on macro, but you have to go several layers deep to get to it.

Thanks for all your work!

**Comment by lanman1** [comment]
Thank you!

**Reply by webkonto** [reply]
That is already implemented. See macro definition and Moonraker information about macros. This question is explicitely described in the docs. You define a macro in moonraker and accordingly one in Klipper. You will get a macro button in klipperscreen where you can set GPIOs. I attached a 12V driver module controled by a GPIO and driving a 12V light chain. Woks perfect. You may even set startup conditions like "Start always on system start" and so on...

**Reply by webkonto** [reply]
printer.cfg
[gcode_macro POWER_OFF_LED]
gcode:
  {action_call_remote_method("set_device_power",
                             device="led_stripe",
                             state="off")}

[gcode_macro POWER_ON_LED]
gcode:
  {action_call_remote_method("set_device_power",
                             device="led_stripe",
                             state="on")}


moonraker.conf
[power led_stripe] <-- This is the name of your "project"
type: gpio
pin: gpio17    <-- Here you enter the wanted GPIO
off_when_shutdown: True   <-- switch light off on shutdown
initial_state: on  <--switch light on on boot


This should do! Dont't grill your Pi configuring the GPIOs
Maybe something like this makes sense...
https://www.thingiverse.com/thing:5238159
